src/checker.rs

```diff
@@ -47,17 +47,35 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
         return false;
     }
 
-    let mut words = trimmed.split_whitespace();
+    let words = trimmed.split_whitespace();
     let word_count = words.clone().count();
     if word_count < rules.min_word_count
         || word_count > rules.max_word_count
-        || words.any(|word| rules.disallowed_words.contains(
+        || words.clone().any(|word| rules.disallowed_words.contains(
              &word.trim_matches(|c: char| !c.is_alphabetic()).to_lowercase()
            ))
     {
         return false;
     }
 
+    if !rules.stem_separator_regex.is_empty() && !(rules.stem_separator_regex == "[]") {
```

Comment:
```suggestion
    if !rules.stem_separator_regex.is_empty() && rules.stem_separator_regex != "[]" {
```

That being said, I don't think we should specifically check for `[]`. That is invalid is per:

```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    []
    ^^
error: unclosed character class
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)', src/checker.rs:62:68
```

and IMHO should lead to an actual error instead of us swallowing that case silently. The same goes for any other invalid regex. WDYT?

Reply:
I added it to hide user errors for better UX. People might set it to [] and might not know what the error means.

The whole experience is based on experimentation, adding/removing etc. Such user mistakes might happen.

But it is your decision...

Reply:
Removed that case as suggested.